consensus/src/ordered_broadcast/mod.rs

```diff
@@ -577,12 +597,76 @@ mod tests {
                 &mut registrations,
                 &mut automatons.lock().unwrap(),
                 &mut committers,
-                Duration::from_millis(100),
                 Duration::from_secs(5),
                 |i| i % 10 == 0,
             );
 
-            await_committers(context.with_label("committer"), &committers, 100).await;
+            await_committers(context.with_label("committer"), &committers, (100, 111)).await;
+        });
+    }
+
+    #[test_traced]
+    fn test_updated_epoch() {
+        let num_validators: u32 = 4;
+        let quorum: u32 = 3;
+        let (runner, mut context, _) = Executor::timed(Duration::from_secs(60));
+        let (identity, mut shares_vec) =
+            ops::generate_shares(&mut context, None, num_validators, quorum);
+        shares_vec.sort_by(|a, b| a.index.cmp(&b.index));
+
+        runner.start(async move {
+            // Setup network
+            let (mut oracle, validators, pks, mut registrations) = initialize_simulation(
+                context.with_label("simulation"),
+                num_validators,
+                &mut shares_vec,
+            )
+            .await;
+            let automatons = Arc::new(Mutex::new(
+                BTreeMap::<PublicKey, mocks::Automaton<PublicKey>>::new(),
+            ));
+            let mut committers =
+                BTreeMap::<PublicKey, mocks::CommitterMailbox<Ed25519, Sha256Digest>>::new();
+            let monitors = spawn_validator_engines(
+                context.with_label("validator"),
+                identity.clone(),
+                &pks,
+                &validators,
+                &mut registrations,
+                &mut automatons.lock().unwrap(),
+                &mut committers,
+                Duration::from_secs(1),
+                |_| false,
+            );
+
+            // Perform some work
+            await_committers(context.with_label("committer"), &committers, (100, 111)).await;
+
+            // Simulate partition by removing all links.
+            link_validators(&mut oracle, &pks, Action::Unlink, None).await;
+            context.sleep(Duration::from_secs(30)).await;
+
+            // Get the maximum height from all committers.
+            let max_height = get_max_height(&mut committers).await;
+
+            // Update the epoch
+            for monitor in monitors.values() {
+                monitor.update(112);
+            }
+
+            // Heal the partition by re-adding links.
+            let link = Link {
+                latency: 10.0,
+                jitter: 1.0,
+                success_rate: 1.0,
+            };
+            link_validators(&mut oracle, &pks, Action::Link(link), None).await;
+            await_committers(
+                context.with_label("committer"),
+                &committers,
+                (max_height + 100, 112),
```

Comment:
I modified `await_committers` so that we can assert the epoch of the latest confirmed chunk has increased.